src/components/binding/binding-panel.js

```diff
@@ -213,13 +187,44 @@ genotet.BindingPanel.prototype.updateTracks = function() {
       this.container.find('#genes #track-' + index).remove();
     }
   }
+
+  this.signal('loadBindingList');
+
+  this.container.find('#genes .glyphicon-remove')
+    .css('display', this.data.tracks.length == 1 ? 'none' : '');
+};
+
+/**
+ * Updates the track list for panel.
+ */
+genotet.BindingPanel.prototype.updateTrackList = function() {
+  var fileNames = genotet.data.bindingFiles.map(function(dataInfo) {
+    return {
+      id: dataInfo.fileName,
+      text: dataInfo.gene + ' (' + dataInfo.fileName + ')'
+    };
+  });
   this.data.tracks.forEach(function(track, index) {
     var ui = this.container.find('#genes #track-' + index);
+
     if (!ui.length) {
```

Comment:
This is called update tracklists.   it would make better sense if update not handles addition edit removal or update handles everything including addition and removal. In this case you're handling only addition but not removal. So it's a little bit strange. Besides I didn't figure out where you handle the track removals for the panel.